### PR TITLE
Switch to llvm-mingw for faster Windows CI

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -791,7 +791,7 @@ test_libnimbus_lc: libnimbus_lc.a
 			if (( $${WITH_UBSAN:-0} )); then \
 				echo "MINGW cannot find -lubsan." && exit 1; \
 			fi; \
-			gcc -D__DIR__="\"beacon_chain/libnimbus_lc\"" --std=c17 -Wall -Wextra -pedantic -Werror -pedantic-errors -flto -Wl,--stack,0x0000000000800000 -o build/test_libnimbus_lc -D_CRT_SECURE_NO_WARNINGS beacon_chain/libnimbus_lc/test_libnimbus_lc.c build/libnimbus_lc.a "$${EXTRA_FLAGS[@]}"; \
+			gcc -D__DIR__="\"beacon_chain/libnimbus_lc\"" --std=c17 -Wall -Wextra -pedantic -Werror -pedantic-errors -flto -Wno-nullability-extension -Wl,--stack,0x0000000000800000 -o build/test_libnimbus_lc -D_CRT_SECURE_NO_WARNINGS beacon_chain/libnimbus_lc/test_libnimbus_lc.c build/libnimbus_lc.a "$${EXTRA_FLAGS[@]}"; \
```
Comment (arnetheduck) [inline Makefile:793]: shouldn't this ubsan run on clang too?

**Comment by arnetheduck** [comment]
switching tests means that we must also switch release binary production (so  that the unit test runs are representative of what ends up getting compiled on release / nightly

**Comment by jangko** [comment]
Release binary is build using cross compilation tools from linux. Don't know if there exists such tool from llvm.

File: Makefile (hunk 1 above)
Comment (jangko) [reply Makefile:793]: fixed

**Pull request merged**